src/ngRoute/route.js

```diff
@@ -160,7 +171,7 @@ function $RouteProvider() {
    */
   this.when = function(path, route) {
     //copy original route object to preserve params inherited from proto chain
-    var routeCopy = angular.copy(route);
+    var routeCopy = createShallowCopy(route);
```

Comment:
Isn't this another case where we should split the shared function into its own file so that we can reuse it in multiple modules?